plugins/optimization-detective/debug.php

```diff
@@ -0,0 +1,226 @@
+<?php
+/**
+ * Debug helpers used for Optimization Detective.
+ *
+ * @package optimization-detective
+ * @since n.e.x.t
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Registers tag visitors.
+ *
+ * @since n.e.x.t
+ *
+ * @param OD_Tag_Visitor_Registry $registry Tag visitor registry.
+ */
+function od_debug_register_tag_visitors( OD_Tag_Visitor_Registry $registry ): void {
+	if ( ! current_user_can( 'customize' ) && ! wp_is_development_mode( 'plugin' ) ) {
+		return;
+	}
```

Comment:
I see this logic appearing in multiple places. Probably makes sense to incorporate into a new helper function? I understand the purpose of this is to restrict access to the debug data, but also note that https://github.com/WordPress/performance/pull/1788 will allow for optimizations to apply by default to admins. I don't think the `&& ! wp_is_development_mode( 'plugin' )` condition is needed?

Reply:
> I see this logic appearing in multiple places. Probably makes sense to incorporate into a new helper function?

👍 

> I understand the purpose of this is to restrict access to the debug data

Yes, basically I want to add the admin bar menu item (and any debug markup that is needed for that) only for admins.